test/YesSql.Tests/CoreTests.cs

```diff
@@ -2301,6 +2301,39 @@ public async Task ShouldGetDynamicById()
             }
         }
 
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(5)]
+        [Theory]
+        public async Task ShouldReturnObjectsByIdsInCorrectOrder(int numberOfItems)
+        {
+            var circleIds = new List<int>();
+
+            using (var session = _store.CreateSession())
+            {
+                for (var i = 0; i < numberOfItems; i++)
+                {
+                    var circle = new Circle
+                    {
+                        Radius = 10
+                    };
+                    session.Save(circle);
+                    circleIds.Add(circle.Id);
+                }
+
+                await session.CommitAsync();
+            }
+
+            using (var session = _store.CreateSession())
+            {
+                circleIds.Reverse();
```

Comment:
The test is wrong, I don't think `Reverse()` mutates the callee. It returns a new collection instead. So your test is still testing the same initial order.

```suggestion
                circleIds = circleIds.Reverse();
```

Reply:
Add `.ToList()` to that

Reply:
it mutates the list: https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.reverse?view=netframework-4.7.2

Reply:
i was surprised too :)